nutils/util.py

```diff
@@ -419,4 +419,30 @@ def readtext(path):
 
   raise TypeError('readtext requires a path-like or file-like argument')
 
+def binaryfile(path):
+  '''Open file for binary reading
+
+  Args
+  ----
+  path: :class:`os.PathLike`, :class:`str` or :class:`io.BufferedIOBase`
+      Path-like or file-like object pointing to the data to be read.
+
+  Returns
+  -------
+  :
+      Context that returns a :class:`io.BufferedReader` upon entry.
+  '''
+
+  if isinstance(path, pathlib.Path):
+    return path.open('rb')
+
+  if isinstance(path, str):
+    return open(path, 'rb')
+
+  if isinstance(path, io.BufferedIOBase):
+    return contextlib.nullcontext(path) if hasattr(contextlib, 'nullcontext') \
+      else contextlib.contextmanager(iter)([path]) # Python <= 3.6
+
+  raise TypeError('readtext requires a path-like or file-like argument')
```

Comment:
The error message should read `binaryfile` instead of `readtext`.

Reply:
fixed